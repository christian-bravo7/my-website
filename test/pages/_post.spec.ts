import Post from '@/pages/_post.vue';
import PortalVue from 'portal-vue';
import VueMeta from 'vue-meta';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();
localVue.use(PortalVue);
localVue.use(VueMeta, { keyName: 'head' });

describe('single post page', () => {
  it('should render page correctly', () => {
    const wrapper = mount(Post, {
      localVue,
      mocks: {
        $route: {
          path: 'localhost',
        },
      },
      data: () => ({
        article: {
          Post: 'hello-world',
          title: 'Hello world',
          description: 'Welcome to my first post of the many that I hope to share us ❤️',
          minutes: 2,
          image: 'https://images.pexels.com/photos/2646531/pexels-photo-2646531.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
          date: [2020, 10, 24],
        },
      }),
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should display meta-tag title', () => {
    const wrapper = mount(Post, {
      localVue,
      mocks: {
        $route: {
          path: 'localhost',
        },
      },
      data: () => ({
        article: {
          Post: 'hello-world',
          title: 'Hello world',
          description: 'Welcome to my first post of the many that I hope to share us ❤️',
          minutes: 2,
          image: 'https://images.pexels.com/photos/2646531/pexels-photo-2646531.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
          date: [2020, 10, 24],
        },
      }),
    });

    const title = wrapper.vm.$meta().refresh().metaInfo.title;

    expect(title).toBe('Hello world');
  });
});
