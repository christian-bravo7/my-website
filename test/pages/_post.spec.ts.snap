// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single post page should render page correctly 1`] = `
<section>
  <div class="v-portal" style="display: none;"></div>
  <div class="relative | container mx-auto">
    <div class="w-full | py-12 sm:py-20 px-4 sm:px-12 | shadow-lg | rounded-2xl | bg-gray-50 dark:bg-blue-900 | transform -translate-y-20">
      <div class="max-w-7xl w-full | mx-auto"><span document="[object Object]" class="blog-content"></span></div>
    </div>
  </div>
</section>
`;
